pkg/hooks/types.go

```diff
@@ -0,0 +1,112 @@
+package hooks
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/square/p2/pkg/logging"
+	"github.com/square/p2/pkg/types"
+)
+
+// HookType represents the stage in the Hook lifecycle
+type HookType string
+
+func (hookType HookType) String() string {
+	return string(hookType)
+}
+
+var (
```

Comment:
so, while yo'ure moving these - any chance of making these `const`s? Doesn't seem we'd want to change them.